substrate/frame/bags-list/src/lib.rs

```diff
@@ -423,19 +497,64 @@ impl<T: Config<I>, I: 'static> SortedListProvider<T::AccountId> for Pallet<T, I>
 	}
 
 	fn on_insert(id: T::AccountId, score: T::Score) -> Result<(), ListError> {
-		List::<T, I>::insert(id, score)
+		match LockOrder::<T, I>::get() {
+			// insert node at the tail of the list to not affect the order of the list while
+			// ordering lock is set.
+			Some(_) => List::<T, I>::insert_force_lowest(id, score),
+			None => List::<T, I>::insert(id, score),
+		}
 	}
 
 	fn get_score(id: &T::AccountId) -> Result<T::Score, ListError> {
 		List::<T, I>::get_score(id)
 	}
 
 	fn on_update(id: &T::AccountId, new_score: T::Score) -> Result<(), ListError> {
-		Pallet::<T, I>::do_rebag(id, new_score).map(|_| ())
+		match LockOrder::<T, I>::get() {
```

Comment:
This LockOrder condition makes the code bit harder to read. May be there is a way to simplify this.